Lib/unittest/case.py

```diff
@@ -63,24 +64,65 @@ def testPartExecutor(self, test_case, subTest=False):
         except _ShouldStop:
             pass
         except:
-            exc_info = sys.exc_info()
             if self.expecting_failure:
-                self.expectedFailure = exc_info
+                self.expectedFailure = sys.exc_info()
             else:
                 self.success = False
                 if subTest:
-                    self.result.addSubTest(test_case.test_case, test_case, exc_info)
+                    self.result.addSubTest(test_case.test_case, test_case, sys.exc_info())
                 else:
-                    _addError(self.result, test_case, exc_info)
-            # explicitly break a reference cycle:
-            # exc_info -> frame -> exc_info
-            exc_info = None
+                    _addError(self.result, test_case, sys.exc_info())
+                if self.debug:
+                    _handle_debug_exception(
+                        self.debug,
+                        on_crash=not subTest and
+                        (lambda: self.pm_frame_holds.append(_AutoDelRunner(self.pm_cleanup))))
```

Comment:
These two lines are a bit dense

Reply:
So I'd format it `black` style when ok? :
```diff

                     _handle_debug_exception(
                         self.debug,
-                        on_crash=not subTest and
-                        (lambda: self.pm_frame_holds.append(_AutoDelRunner(self.pm_cleanup))))
+                        on_crash=not subTest
+                        and (
+                            lambda: self.pm_frame_holds.append(
+                                _AutoDelRunner(self.pm_cleanup)
+                            )
+                        ),
+                    )
         else:

```

Reply:
I don’t think black gives nice results in these cases.  I would extract the test and lambda and move them before the `_handle_debug_exception(` line.

Reply:
So I'd go like this - or could insert another line break before `_AutoDel...` :
```diff
                 if self.debug:
-                    _handle_debug_exception(
-                        self.debug,
-                        on_crash=not subTest and
-                        (lambda: self.pm_frame_holds.append(_AutoDelRunner(self.pm_cleanup))))
+                    if subTest:
+                        on_crash = None
+                    else:
+                        def on_crash():
+                            self.pm_frame_holds.append(_AutoDelRunner(self.pm_cleanup))
+                    _handle_debug_exception(self.debug, on_crash=on_crash)
         else:
```

Reply:
It could be:
```python
    if self.debug:
        on_crash = None
        if not subTest:
            on_crash = lambda: self.pm_frame_holds.append(_AutoDelRunner(self.pm_cleanup))
```

Reply:
Ok.